Diagnostics/HealthChecker/DataCollection/ExchangeInformation/IISInformation/Get-IISWebSite.ps1

```diff
@@ -42,20 +42,87 @@ function Get-IISWebSite {
+        $webConfigContentXml = $null
         $validWebConfig = $false
+        $customHeaderHstsObj = [PSCustomObject]@{
+            enabled             = $false
+            "max-age"           = 0
+            includeSubDomains   = $false
+            preload             = $false
+            redirectHttpToHttps = $false
+        }
+        $customHeaderHsts = $null
 
         if ($webConfigExists) {
             $webConfigContent = (Get-Content $configurationFilePath -Raw).Trim()
 
             try {
-                [xml]$webConfigContent | Out-Null
+                $webConfigContentXml = [xml]$webConfigContent
                 $validWebConfig = $true
             } catch {
                 # Inside of Invoke-Command, can't use Invoke-CatchActions
                 Write-Verbose "Failed to convert IIS web config '$configurationFilePath' to xml. Exception: $($_.Exception)"
             }
         }
 
+        if ($validWebConfig) {
+            <#
+                HSTS configuration can be done in different ways:
+                Via native HSTS control that comes with IIS 10.0 Version 1709.
+                See: https://learn.microsoft.com/iis/get-started/whats-new-in-iis-10-version-1709/iis-10-version-1709-hsts
+                The native control stores the HSTS configuration attributes in the <hsts> element which can be found under each <site> element.
+                These settings are returned when running the Get-WebSite cmdlet and there is no need to prepare the data as they are ready for use.
+
+                Via customHeader configuration (when running IIS older than the version mentioned before where the native HSTS config is not available
+                or when admins prefer to do it via customHeader as there is no requirement to do it via native HSTS control instead of using customHeader).
+                HSTS via customHeader configuration are stored under httpProtocol.customHeaders element. As we get the content in the previous
+                call, we can simply use these data to extract the customHeader with name Strict-Transport-Security (if exists) and can then prepare
+                the data for further processing.
+                The following code searches for a customHeader with name Strict-Transport-Security. If we find the header, we then extract the directive
+                and return them as PSCustomObject. We're looking for the following directive: max-age, includeSubDomains, preload, redirectHttpToHttps
+            #>
+            $customHeaderHsts = ($webConfigContentXml.configuration.'system.webServer'.httpProtocol.customHeaders.add | Where-Object {
+                ($_.name -eq "Strict-Transport-Security")
+                }).value
+            if ($null -ne $customHeaderHsts) {
+                Write-Verbose "Hsts via custom header configuration detected"
+                $customHeaderHstsObj.enabled = $true
+                $maxAgeIndex = $customHeaderHsts.IndexOf("max-age=")
```

Comment:
Need to determine how to handle if value that was entered was not in the correct case sensitive based off the doc. 

1. Does the setting still work and how do we verify? 
2. Depending on the answer to # 1, then we need to determine how to handle this for `IndexOf` as that is case sensitive. If the IIS setting doesn't care, then we can do a simple `ToLower()` on the `string` and be okay. But if it is case sensitive, then we need to have an additional parameter stating that it is misconfigured due to case sensitive.

Reply:
![image](https://github.com/microsoft/CSS-Exchange/assets/22776718/258d9373-7cc0-40e6-a708-99ba6018b1c9)

Reply:
Will be fixed with next commit.